src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseField.java

```diff
@@ -140,6 +144,46 @@ public void propagateACLsToNestedFields() {
         propagateACLsToNestedFields(this, this.accessControlLists);
     }
 
+    @JsonIgnore
+    public FieldType getFieldTypeByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            return this.getFieldType();
+        } else {
+            CaseField caseFieldByPath = this.findNestedElementByPath(path);
+            return caseFieldByPath.getFieldType();
+        }
+    }
+
+    @JsonIgnore
+    public CaseField findNestedElementByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            throw new RuntimeException(format("Invalid blank element path for field %s.", this.id));
+        }
+        if (this.getFieldType().getChildren().isEmpty()) {
+            throw new RuntimeException(format("CaseField %s has no nested elements.", this.id));
+        }
+        List<String> pathElements = Arrays.stream(path.trim().split("\\.")).collect(Collectors.toList());
+
+        return reduce(this.getFieldType().getChildren(), pathElements);
+    }
+
+    @JsonIgnore
+    private CaseField reduce(List<CaseField> caseFields, List<String> pathElements) {
+        String head = pathElements.get(0);
+        if (pathElements.size() == 1) {
+            return caseFields.stream().filter(e -> e.getId().equals(head)).findFirst()
+                .orElseThrow(() -> new RuntimeException(format("Nested element not found for %s", head)));
+        } else {
+            CaseField caseField = caseFields.stream().filter(e -> e.getId().equals(head)).findFirst()
```

Comment:
this block of code is duplicated with the above. Can we remove this duplication?

Reply:
Done